File: update-cover-list.py
#!/usr/bin/env python3
"""
扫描 assets/cover/ 里的图片，生成 list.json。
用法：python3 update-cover-list.py
把要轮播的图放进 assets/cover/ 后运行即可，无需改配置。
"""
import os
import json

dir_path = os.path.join(os.path.dirname(__file__), "assets", "cover")
exts = {".jpg", ".jpeg", ".png", ".gif", ".webp"}
names = []

if os.path.isdir(dir_path):
    for f in os.listdir(dir_path):
        if f.startswith("."):
            continue
        if os.path.splitext(f)[1].lower() in exts:
            names.append(f)
    names.sort()

out_path = os.path.join(dir_path, "list.json")
os.makedirs(dir_path, exist_ok=True)
with open(out_path, "w", encoding="utf-8") as f:
    json.dump(names, f, indent=2, ensure_ascii=False)

print("Wrote assets/cover/list.json:", len(names), "files:", names)

File: update-cover-list.js
#!/usr/bin/env node
/**
 * 扫描 assets/cover/ 里的图片，生成 list.json。
 * 用法：node update-cover-list.js
 * 之后把 assets/cover/ 和 list.json 一起 git add 提交即可。
 */
const fs = require('fs');
const path = require('path');

const dir = path.join(__dirname, 'assets', 's1-ceramics', 'cover');
const exts = ['.jpg', '.jpeg', '.png', '.gif', '.webp'];
let names = [];

try {
  if (fs.existsSync(dir)) {
    names = fs.readdirSync(dir)
      .filter(f => {
        const ext = path.extname(f).toLowerCase();
        return exts.includes(ext) && !f.startsWith('.');
      })
      .sort();
  }
} catch (e) {
  console.error(e);
}

const outPath = path.join(dir, 'list.json');
fs.mkdirSync(dir, { recursive: true });
fs.writeFileSync(outPath, JSON.stringify(names, null, 2));
console.log('Wrote assets/cover/list.json:', names.length, 'files:', names);
